File: cpp/1935.maximum-number-of-words-you-can-type.cpp
class Solution {
public:
    int canBeTypedWords(string text, string brokenLetters) {
	    unordered_set<char> broken;
	    for(auto letter:brokenLetters)broken.insert(letter);
	    int res = 0;
	    text +=" ";
	    string current = "";
	    bool has_invalid = false;
	    for(auto letter:text){
		    if(letter == ' '){
			    if(!has_invalid) res++;
			    current = "";
			    has_invalid=false;
		    }else current += letter;
		    if(broken.count(letter)) has_invalid=true;
	    }
	    return res;
    }
};

File: python/1935.maximum-number-of-words-you-can-type.py
class Solution:
    def canBeTypedWords(self, text: str, brokenLetters: str) -> int:
        res = 0
        for word in text.split():
            for letter in word:
                if letter in brokenLetters:
                    res+=1
                    break
        return len(text.split())-res
